### PR TITLE
Update demo model paths

### PR DESCRIPTION
**Pull request by ibelem**
The models of 4 WebNN Developer Preview demos have been published into https://huggingface.co/microsoft, update the model paths in README.md and demos source code. 

@Adele101 PTAL

### CODE REVIEW

File: demos/whisper-base/main.js [hunk 1]
```diff
@@ -672,7 +672,7 @@ const main = async () => {
   try {
     context = new AudioContext({ sampleRate: kSampleRate });
     const whisper_url = location.href.includes("github.io")
-      ? "https://huggingface.co/onnxruntime-web-temp/demo/resolve/main/whisper-base/"
+      ? "https://huggingface.co/microsoft/whisper-base-webnn/resolve/main/"
```
Comment (fdwr) [inline demos/whisper-base/main.js:675]: Curiously when I paste "https://huggingface.co/microsoft/whisper-base-webnn/resolve/main/" into the browser, I get 404? (though I get "entry not found" with the old URL too, and so something more is happening here)

**Review by fdwr — APPROVED** [review]
👍 Thanks B.

File: demos/whisper-base/main.js (hunk 1 above)
Comment (ibelem) [reply demos/whisper-base/main.js:675]: @fdwr 

We need to add detailed file names after the whisper_url, e.g.

- https://huggingface.co/microsoft/whisper-base-webnn/resolve/main/whisper_base_encoder_lm_fp16_layernorm.onnx
- https://huggingface.co/onnxruntime-web-temp/demo/resolve/main/whisper-base/whisper_base_encoder_lm_fp16_layernorm.onnx

Comment (fdwr) [reply demos/whisper-base/main.js:675]: Yeah, we definitely need to concatenate the final filenames afterward for the full URL. I just figured there would be a directory listing or something. (resolving this lingering open tab)

Comment (ibelem) [reply demos/whisper-base/main.js:675]: @fdwr If you want the directory list, https://huggingface.co/microsoft/whisper-base-webnn/tree/main will work,  the https://huggingface.co/microsoft/whisper-base-webnn/resolve/main/ +filenames for download or via fetch() API in JS only.

`tree` vs `resolve`

Comment (fdwr) [reply demos/whisper-base/main.js:675]: 🫡

**Pull request merged**